### ISSUE TITLE
Mutable default constructor argument values in PunktTrainer and PunktSentenceTokenizer

### ISSUE THREAD

**Issue by brox** [nltk/nltk#1108]
When creating several `nltk.tokenize.punkt.PunktTrainer` instances they end up interfering with each other:

```
>>> from nltk.tokenize.punkt import PunktTrainer
>>> x = PunktTrainer()
>>> x._params.ortho_context
defaultdict(<type 'int'>, {})
>>> x.train('Tokenization is difficult, e.g. when we have abbreviations.')
>>> x._params.ortho_context
defaultdict(<type 'int'>, {u'we': 32, u'e.g.': 32, u'is': 32, u'when': 64, u'tokenization': 8, u'have': 32, u'abbreviations': 32, u'difficult': 32})
>>> y = PunktTrainer()
>>> y._params.ortho_context
defaultdict(<type 'int'>, {u'we': 32, u'e.g.': 32, u'is': 32, u'when': 64, u'tokenization': 8, u'have': 32, u'abbreviations': 32, u'difficult': 32})
```

This is because `nltk.tokenize.punkt.PunktBaseClass` instantiates `PunktParameters` as a constructor default argument. The `PunktParameters` instance will then be reused when creating new `PunktTrainer` objects:

```
def __init__(self, lang_vars=PunktLanguageVars(), token_cls=PunktToken,
             params=PunktParameters()):
```

`nltk.tokenize.punkt.PunktSentenceTokenizer` has the same issue.

**Comment by stevenbird** [comment]
Thanks for the bug report @brox.

@arthurdarcet or @jnothman – do you have any time to look into this please?

**Comment by brox** [comment]
Thanks @stevenbird.

I'd be happy to submit a pull request after the weekend to save you some time.

**Comment by alvations** [comment]
This should be resolved in #2067 =)
